plugins/modules/operatingsystem.py

```diff
@@ -211,12 +211,12 @@ def main():
         entity = None
         # If we have a description, search for it
         if 'description' in module_params and module_params['description'] != '':
-            search_string = 'description="{0}"'.format(module_params['description'])
+            search_string = 'description="{0}" or title="{0}"'.format(module_params['description'])
```

Comment:
What if two descriptions share the same 255 first characters?
Should we not search for description first and then fallback to title?

Reply:
You mean OS1 has description "longlong255" and OS2 has description "longlong255andmore"? That's not possible:

```irb
irb(main):020:0> os1 = Operatingsystem.new(name: 'test1', major: 8, description: 'AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA
AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA')
=> #<Operatingsystem id: nil, major: "8", name: "test1", minor: "", nameindicator: nil, created_at: nil, updated_at: nil, release_name: nil, type: nil, description: "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA...", pass...
irb(main):021:0> os1.save
=> true
irb(main):022:0> os2 = Operatingsystem.new(name: 'test2', major: 8, description: 'AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAandmore')
=> #<Operatingsystem id: nil, major: "8", name: "test2", minor: "", nameindicator: nil, created_at: nil, updated_at: nil, release_name: nil, type: nil, description: "AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA...", pass...
irb(main):023:0> os2.save
=> false
```

The second OS fails validation, as the title needs to be unique (but isn't)

Reply:
So we only need to search by "desc or title" because the user could provide the full _or_ shortened version?

Reply:
In reality we could drop the whole `desc or title` search alltogether, and go by `name,major,minor` only, but that'd be a breaking change and I wanted to avoid that.

And if we keep `desc or title`, it needs to be both for the reasons you stated, yes.

Reply:
And for the uniqueness you already stated, it should be impossible to create a description that matches the `<name> <major>.<minor>` of a different OS. Also i like the convenience we provide here.